src/Servers/Kestrel/Core/src/Internal/HttpConnection.cs

```diff
@@ -371,8 +371,12 @@ private void CloseUninitializedConnection(ConnectionAbortedException abortReason
 
             _context.ConnectionContext.Abort(abortReason);
 
-            _adaptedTransport.Input.Complete();
-            _adaptedTransport.Output.Complete();
+            // TODO this probably should check if Https is available?
+            if (_context.ConnectionAdapters.Count > 0)
```

Comment:
@davidfowl I'm not if we need this check still?

Reply:
Nuke all of the connection adapter code.